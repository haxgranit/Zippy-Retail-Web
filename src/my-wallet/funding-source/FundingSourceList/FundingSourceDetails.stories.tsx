import { ComponentStory, ComponentMeta } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';
import React from 'react';
import FundingSourceList from './FundingSourceList';

export default {
  title: 'my-wallet/funding-source/FundingSourceList',
  component: FundingSourceList,
} as ComponentMeta<typeof FundingSourceList>;

const Template: ComponentStory<typeof FundingSourceList> = () => (
  <BrowserRouter>
    <FundingSourceList />
  </BrowserRouter>
);

export const Default = Template.bind({});
Default.args = {};
Default.parameters = {};
