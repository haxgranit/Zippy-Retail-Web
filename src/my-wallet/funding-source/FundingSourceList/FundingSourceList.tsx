import { useMsal } from '@azure/msal-react';
import { useEffect, useState } from 'react';
import { Button } from 'react-bootstrap';
import { useNavigate } from 'react-router-dom';
import Api, { FundingSource } from '../../../api';
import PageContainer from '../../../common/PageContainer';

export default function FundingSourceList() {
  const navigate = useNavigate();

  const { instance, accounts } = useMsal();
  const [fundingSources, setFundingSources] = useState<FundingSource[]>([]);

  useEffect(() => {
    new Api(instance, accounts[0])
      .listFundingSources()
      .then((result:FundingSource[]) => {
        setFundingSources(result);
      })
      .catch((error) => console.log('error', error));
  }, []);

  return (
    <PageContainer title="Your Wallet" subTitle="Made Fun With Zippy!" backdropImage="background-funding-image">
      <div className="fundingTitle">
        Funding Source
      </div>

      <div>
        <div className="bankWrapper">
          <div style={{ display: 'flex', justifyContent: 'center', alignItems: 'center' }}>
            <i className="zippy-cash-icon zc-bank" />
            {' '}
            <p className="bankTitle">Registered Banks</p>
          </div>
          <div
            onKeyDown={() => { }}
            role="button"
            tabIndex={0}
            onClick={() => navigate('/my-wallet/funding-source/add-funding-source')}
            className="addBtn"
          >
            +Add
          </div>
        </div>

        { fundingSources.map((source:FundingSource) => (
          <div className="bankDetails">
            <div style={{ display: 'flex', justifyContent: 'center', alignItems: 'flex-start' }}>
              <input type="radio" />
              <div style={{ marginLeft: 11 }}>
                <p className="bankName">{source.displayName}</p>
                <p className="accNo">
                  Acc. Number:
                  {' '}
                  {source.bankAccount?.accountNumber}
                </p>
              </div>
            </div>
            <div className="editWrap">
              <i className="zippy-cash-icon zc-cross" />
              <div
                className="editBtn"
                onKeyDown={() => { }}
                role="button"
                tabIndex={0}
                onClick={() => navigate(`/my-wallet/funding-source/edit-funding-source/${source.id}`)}
              >
                Edit

              </div>
            </div>
          </div>
        ))}
      </div>
      <div className="action">
        <Button
          className="zippy-btn"
        >
          Send money
        </Button>
      </div>
    </PageContainer>
  );
}
