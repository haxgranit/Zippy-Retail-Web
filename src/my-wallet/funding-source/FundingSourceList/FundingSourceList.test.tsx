import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import FundingSourceList from './FundingSourceList';

const ReactTestRenderer = require('react-test-renderer');

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useNavigate: () => mockedUsedNavigate,
}));

describe('FundingSourceList Component', () => {
  it('matches the snapshot', () => {
    const tree = ReactTestRenderer.create(
      <BrowserRouter>
        <FundingSourceList />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
