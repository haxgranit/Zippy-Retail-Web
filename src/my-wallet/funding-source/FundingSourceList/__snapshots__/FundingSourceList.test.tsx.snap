// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FundingSourceList Component matches the snapshot 1`] = `
<div
  className=""
>
  <div
    className="zippy-cash-container-wrap"
  >
    <div
      className="backdrop backdrop-image-4"
    />
    <div
      className="zippy-cash-container"
    >
      <div
        className="zippy-cash-container-header"
      >
        <h1
          className="title"
        >
          Your Wallet
        </h1>
        <h2
          className="catch"
        >
          Made Fun With Zippy!
        </h2>
      </div>
      <div
        className="zippy-cash-container-body"
      >
        <div
          className="funding-title"
        >
          Funding Source
        </div>
        <div
          style={
            Object {
              "height": 380,
            }
          }
        >
          <div
            className="bank-wrapper"
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "center",
                }
              }
            >
              <i
                className="zippy-cash-icon zc-bank"
              />
               
              <p
                className="bank-title"
              >
                Registered Banks
              </p>
            </div>
            <div
              className="add-btn"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              +Add
            </div>
          </div>
        </div>
        <div
          className="action"
        >
          <button
            className="zippy-btn btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Send money
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
